cli/main.go

```diff
@@ -93,6 +107,15 @@ func start(_ context.Context, cmd *cli.Command) error {
 		log.Error("no relay passed the health-check!")
 	}
 
+	if opts.PrometheusListenAddr > 0 && opts.PrometheusListenAddr <= math.MaxUint16 {
```

Comment:
Doesn't look like listenAddr has any validation on, nor any existing precedent for pointer values coming from the flags, so thought this would be somewhat sensible given any non-0 value would pass and initialize the server